hail/python/hailtop/utils/utils.py

```diff
@@ -375,15 +384,11 @@ async def _shutdown(self) -> None:
             return
 
         # shut down the pending tasks
-        tasks = []
         for _, t in self._pending.items():
             if not t.done():
-                t.cancel()
-            tasks.append(t)
-        self._pending = None
+                await cancel_task(t)
```

Comment:
One more note, you can't `await` inside the loop over pending tasks, because another coroutine could add new task to _pending, too (that is, _pending can be modified in two ways).